app/models/concerns/order_status.rb

```diff
@@ -93,6 +92,30 @@ module OrderStatus # rubocop:disable Metrics/ModuleLength
       event :close do
         transition received: :closed
       end
+
+      event :cancel do
+        old_status = ""
+
+        before do
+          old_status = status
+        end
+
+        transition all - [:canceled, :rejected] => :canceled
+
+        after do
+          case old_status
+          when "filled", "shipped", "received", "closed"
+            order_details.each do |order_detail|
+              item = order_detail.item
+              item.mark_event(edit_amount: order_detail.quantity,
+                              edit_method: "add",
+                              edit_reason: "order_canceled_adjustment",
```

Comment:
Please see [config/locales/en.yml](https://github.com/on-site/StockAid/blob/master/config/locales/en.yml#L30-L35) and update that list with this new edit reason, otherwise this will not show up in the inventory history properly.

Reply:
👍